tensorflow_lite_support/c/task/processor/base_options.h

```diff
@@ -0,0 +1,77 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#ifndef TENSORFLOW_LITE_SUPPORT_C_TASK_PROCESSOR_BASE_OPTIONS_H_
+#define TENSORFLOW_LITE_SUPPORT_C_TASK_PROCESSOR_BASE_OPTIONS_H_
+
+#include <stdint.h>
+
+// Defines C Structs for Base Options Shared by all tasks.
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+// Holds settings to configure cpu execution.
+typedef struct TfLiteCpuSettings {
+  // Specifies the number of threads to be used for TFLite
+  // ops that support multi-threading when running inference with CPU.
+  // num_threads should be greater than 0 or equal to -1. Setting num_threads to
+  // -1 has the effect to let TFLite runtime set the value.
+  int num_threads;
+} TfLiteCpuSettings;
+
+// Holds settings to configure TFLite runtime.
+typedef struct TfLiteTflSettings {
+  // Specifies how to configure cpu execution.
+  TfLiteCpuSettings cpu_settings;
+} TfLiteTflSettings;
+
+// Holds settings for one possible acceleration configuration.
+typedef struct TfLiteComputeSettings {
+  // Holds settings to configure TFLite runtime.
+  TfLiteTflSettings tflite_settings;
```

Comment:
TFLiteSettings can be removed. Other fields in ComputeSettings is not used, therefore. Please just expose the everything under TFLiteSettings under ComputeSettings. See the Java API for references. https://github.com/tensorflow/tflite-support/blob/1ad1e531d7cf439c200f9df3ae87009d986a6f52/tensorflow_lite_support/java/src/java/org/tensorflow/lite/task/core/BaseOptions.java

Reply:
Note that, ExternalFile is not added in the Java API yet. But I think what you did here (TfLiteExternalFile) looks good.

Reply:
I have removed all the nested structs in `TfLiteComputeSettings`. In Java, the `num_threads` is in Base Options. But I have maintained `num_threads` as a member of `TfLiteComputeSettings` since that feels more appropriate given the structure of the lite options. Later acceleration related options can also be added to `TfLiteComputeSettings`.